src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/WebserviceController.php

```diff
@@ -0,0 +1,143 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Controller\Admin\Configure\AdvancedParameters;
+
+use PrestaShop\PrestaShop\Core\Form\FormHandlerInterface;
+use PrestaShop\PrestaShop\Core\Grid\Search\WebserviceKeyGridSearchCriteria;
+use PrestaShop\PrestaShop\Core\Webservice\WebserviceCanBeEnabledConfigurationChecker;
+use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;
+use PrestaShopBundle\Security\Annotation\DemoRestricted;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use PrestaShopBundle\Security\Annotation\AdminSecurity;
+
+
+/**
+ * Responsible of "Configure > Advanced Parameters > Webservice" page display
+ *
+ * @todo: add unit tests
+ */
+class WebserviceController extends FrameworkBundleAdminController
+{
+    const CONTROLLER_NAME = 'AdminWebservice';
+
+    /**
+     * Displays the Webservice main page.
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller')~'_')", message="Access denied.")
+     *
+     * @param FormInterface $form
+     * @return Response
+     */
+    public function indexAction(FormInterface $form = null, Request $request)
+    {
+        $form = is_null($form) ? $this->getFormHandler()->getForm() : $form;
+
+        if (Request::METHOD_POST === $request->getMethod()) {
+            if ($request->request->has('form')) {
+                $submittedForm = $request->request->get('form');
+                if (array_key_exists('webservice_configuration', $submittedForm)) {
+                    return $this->processFormAction($request);
```

Comment:
imo, you should have only GET method available in indexAction and processAction in POST, this will reduce the complexity of every action.